assets/src/edit-story/components/fontPicker/pickerContainer.js

```diff
@@ -0,0 +1,217 @@
+`;
+
+const ListContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  flex-wrap: wrap;
+  width: 100%;
+  max-height: 355px;
+  overflow-x: hidden;
+  overflow-y: auto;
+  overscroll-behavior: none auto;
+`;
+
+const List = styled.ul.attrs({ role: 'listbox' })`
+  width: 100%;
+  padding: 5px 0;
+  margin: 0;
+  font-size: 14px;
+  text-align: left;
+  list-style: none;
+`;
+
+const Item = styled.li.attrs({ tabIndex: '0', role: 'option' })`
+  letter-spacing: ${({ theme }) => theme.fonts.label.letterSpacing};
+  padding: 16px;
+  margin: 0;
+  font-family: ${({ theme, fontFamily }) =>
+    fontFamily ? fontFamily : theme.fonts.label.family};
+  font-size: ${({ theme }) => theme.fonts.label.size};
+  line-height: ${({ theme }) => theme.fonts.label.lineHeight};
+  font-weight: ${({ theme }) => theme.fonts.label.weight};
+
+  &:hover {
+    background-color: ${({ theme }) => theme.colors.bg.v12};
+  }
+
+  &:focus {
+    background-color: ${({ theme }) => theme.colors.bg.v12};
+    outline: none;
+  }
+`;
+
+const BoxedTextInput = styled(TextInput)`
+  padding: 6px;
+  border: 1px solid ${({ theme }) => theme.colors.bg.v4};
+  border-radius: 4px;
+  background-color: ${({ theme }) => theme.colors.fg.v1};
+  color: ${({ theme }) => theme.colors.fg.v0};
+`;
+
+const ExpandedTextInput = styled(BoxedTextInput)`
+  flex-grow: 1;
+`;
+
+function FontPickerContainer({
+  handleCurrentValue,
+  value,
+  ariaLabel,
+  options,
+  toggleOptions,
+}) {
+  const {
+    actions: { getMenuFonts },
+  } = useAPI();
+
+  const pickerContainerRef = useRef();
+  const listContainerRef = useRef();
+  const listRef = useRef();
+  const inputRef = useRef();
+
+  const [searchValue, setSearchValue] = useState('');
+  const [menuFonts, setMenuFonts] = useState(null);
+  const [currentIndex, setCurrentIndex] = useState(0);
+  const fileredOptions = useMemo(
+    () =>
+      searchValue && searchValue !== ''
+        ? options.filter(({ name }) =>
+            name.toLowerCase().includes(searchValue.toLowerCase())
+          )
+        : options,
+    [searchValue, options]
+  );
+
+  useEffect(() => {
+    inputRef.current.focus();
+  });
+
+  useLayoutEffect(() => {
+    const listElement = listContainerRef.current;
+
+    const handleScroll = () => {
+      const { scrollTop } = listElement;
+      const currentVisibleIndex = Math.floor(scrollTop / 50);
+      if (currentVisibleIndex !== currentIndex) {
+        setCurrentIndex(currentVisibleIndex);
+      }
+    };
+
+    listElement.addEventListener('scroll', handleScroll, { passive: true });
+
+    return () => {
+      listElement.removeEventListener('scroll', handleScroll);
+    };
+  }, [currentIndex]);
+
+  useEffect(() => {
+    const limitFonts = fileredOptions
+      .slice(currentIndex, currentIndex + 10)
+      .map(({ name }) => name);
+    getMenuFonts(limitFonts).then((result) => {
+      setMenuFonts(result);
+    });
+  }, [currentIndex, getMenuFonts, fileredOptions]);
```

Comment:
Everytime on scrolling, call the api to get 10 (we can make it 6 that visible for now). And there are many api calls and the font picker dropdown actually flickering.. 1000 fonts and maybe I think we can have them all and store the src value as string or inside fonts and that can improve the flickering..